devito/ir/support/syncs.py

```diff
@@ -5,14 +5,18 @@
 from collections import defaultdict
 
 from devito.data import FULL
-from devito.tools import Pickable, filter_ordered
+from devito.tools import Pickable, as_tuple, filter_ordered, frozendict
 from .utils import IMask
 
-__all__ = ['WaitLock', 'ReleaseLock', 'WithLock', 'FetchUpdate', 'PrefetchUpdate',
-           'normalize_syncs']
+__all__ = ['WaitLock', 'ReleaseLock', 'WithLock', 'InitArray', 'SyncArray',
+           'PrefetchUpdate', 'SnapOut', 'SnapIn', 'Ops', 'normalize_syncs']
 
 
-class SyncOp(Pickable):
+class Operation(Pickable):
```

Comment:
Can we name it SyncOperation, we have a whole `operations` module

Reply:
this base class is just a leftover from something else. Good catch, I'm dropping it as it's useless. So we only have SyncOp, as in master